src/value.c

```diff
@@ -358,25 +358,20 @@ int SIValue_ToDouble(SIValue *v, double *d) {
   }
 }
 
-void SIValue_FromString(SIValue *v, char *s) {
-  int numeric = 1;
-  char *c;
-
-  /* Scan string, see if we can find any non-numeric characters in it. */
-  for(c = s; *c != '\0'; c ++) {
-    
-    if(!isdigit(*c) && *c != '.' && *c != '-') {
-      numeric = 0;
-      v->type = T_STRING;
-      break;
-    }
-  }
-
-  if(numeric) {
-    v->type = T_DOUBLE;
+SIValue SIValue_FromString(char *s) {
```

Comment:
I think it would be clearer if the function receives a const.
SIValue SIValue_FromString(const char *s)...